src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTCataLogSearch.java

```diff
@@ -181,18 +190,88 @@ private DataGroup getTableDataFromJson(File jsonFile, TableDef tableDef) throws
+    /**
+     * This method translates the mySql data type to corresponding java data type
+     * @param classType
+     * @return
+     */
+    private Class getDataClass(String classType){
+
+        if (classType.equalsIgnoreCase("double")){
+            return Double.class;
+        }
+        else if (classType.equalsIgnoreCase("float") || classType.equalsIgnoreCase("real") ){
+            return Float.class;
+        }
+        else if (classType.equalsIgnoreCase("int(11)")){
+            return Integer.class;
+        }
+        else if (classType.equalsIgnoreCase("BigInt(20)") ){
+            return Long.class;
+        }
+        else if (classType.equalsIgnoreCase("bit(1)")){
+            return Boolean.class;
+        }
+        else if (classType.equalsIgnoreCase("TINYINT")){
+            return Byte.class;
+        }
+        else if (classType.equalsIgnoreCase("SMALLINT")){
+            return Short.class;
+        }
+        else if (classType.equalsIgnoreCase("string") ) {
+
+            return String.class;
+
+        }
+        else {
+            System.out.println(classType + "is not supported");
+        }
+        return null;
+
+    }
+    private  DataType[] getMetaFromMetatable(DataGroup dataGroup){
+
+        DataType[] dataTypes = new DataType[dataGroup.size()];
+        DataObject[] dataObjects=dataGroup.values().toArray(new DataObject[0]);
+        for (int i=0; i<dataObjects.length; i++){
+            dataTypes[i] = new DataType((String) dataObjects[i].getDataElement("Field"),
+                    getDataClass( (String) dataObjects[i].getDataElement("Type") ) );
+            boolean maybeNull = dataObjects[i].getDataElement("Null").toString().equalsIgnoreCase("yes")?true:false;
+            dataTypes[i] = new DataType( (String) dataObjects[i].getDataElement("Field"),
+                    (String) dataObjects[i].getDataElement("description"),
+                    getDataClass( (String) dataObjects[i].getDataElement("Type")),
+                    DataType.Importance.HIGH,
+                    (String) dataObjects[i].getDataElement("unit"),
+                    maybeNull
+            );
+
+        }
+
+        return dataTypes;
+    }
+    private DataType[] getMeta(TableServerRequest request){
         SearchManager sm = new SearchManager();
 
         try {
-            DataGroupPart dgp = sm.getDataGroup(request);
-            return dgp.getTableDef();
+            FileInfo fi = sm.getFileInfo(request);
+            if (fi == null) {
+                throw new DataAccessException("Unable to get file location info");
+            }
+            if (fi.getInternalFilename() == null) {
+                throw new DataAccessException("File not available");
+            }
+            if (!fi.hasAccess()) {
+                throw new SecurityException("Access is not permitted.");
+            }
+            DataGroup sourceDataGroup = DataGroupReader.readAnyFormat(new File(fi.getInternalFilename()));
```

Comment:
No, it does not work.  This was what I did yesterday around 6PM:

    private TableDef getMeta(TableServerRequest request){
        SearchManager sm = new SearchManager();

        try {
            DataGroupPart dgp = sm.getDataGroup(request);
            DataGroup metaData = dgp.getData();
            return dgp.getTableDef();
        } catch (Exception e) {
            e.getStackTrace();
        }
        return null;
    }

The dataGroup returned contains empty data.  It tells the number of rows and the TableDef but data.  Actually, I don't understand the reason.  I want to ask you about it later.